### PR TITLE
Add lmdb module

### PR DESCRIPTION
**Pull request by benjeffery**
Adding lmdb, as it is a dependency of tsinfer, which I need for a deployment of interactive educational material.

### CODE REVIEW

**Comment by benjeffery** [comment]
> How does the connection to lmdb database works? Does it make HTTP requests or is it lower-level network communications? In the latter case, this would be problematic as opening a socket connection will not work in the browser.

The database is on-disk - similar to sqlite. So no network is needed.

**Comment by ryanking13** [comment]
@benjeffery Could you add some basic tests, so we can make sure it works?

**Review by ryanking13 — APPROVED** [review]
Thanks, @benjeffery!

File: packages/lmdb/test_lmdb.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
+from tempfile import TemporaryDirectory
+
+from pytest_pyodide import run_in_pyodide
+
+
+@run_in_pyodide(packages=["lmdb"])
+def test_lmdb(selenium):
+    import lmdb
```
Comment (ryanking13) [inline packages/lmdb/test_lmdb.py:8]: You need to use `from tempfile import TemporaryDirectory` inside this function.

**Comment by hoodmane** [comment]
Uh oh there seems to be a symbol resolution problem.

**Comment by ryanking13** [comment]
The error message says that it cannot resolve the symbol `pthread_mutexattr_setrobust`, which seems related to threading (Pyodide does not support thread).

I guess there should be a compile flag that disables threading in lmdb? (Maybe you can try flags used here: https://github.com/LMDB/lmdb/blob/3947014aed7ffe39a79991fa7fb5b234da47ad1a/libraries/liblmdb/mdb.c#L313-L339)

**Comment by benjeffery** [comment]
> The error message says that it cannot resolve the symbol `pthread_mutexattr_setrobust`, which seems related to threading (Pyodide does not support thread).
> 
> I guess there should be a compile flag that disables threading in lmdb? Maybe you can try flags used here: https://github.com/LMDB/lmdb/blob/3947014aed7ffe39a79991fa7fb5b234da47ad1a/libraries/liblmdb/mdb.c#L313-L339

Thanks so much for looking into this! I'll try that out.

**Comment by benjeffery** [comment]
> > The error message says that it cannot resolve the symbol `pthread_mutexattr_setrobust`, which seems related to threading (Pyodide does not support thread).
> > I guess there should be a compile flag that disables threading in lmdb? Maybe you can try flags used here: https://github.com/LMDB/lmdb/blob/3947014aed7ffe39a79991fa7fb5b234da47ad1a/libraries/liblmdb/mdb.c#L313-L339
> 
> Thanks so much for looking into this! I'll try that out.

Seems I'd have to make a substantial patch to get this in. Closing for now, thanks for the pointers!

**Pull request closed without merging**